### DISCUSSION TITLE
🚀 Distro 2024.11.1 Released

### DISCUSSION THREAD

**Discussion by Nicogene** [robotology/community#720] [Releases]
# 2024.11.1 Distro Announcement

## Brief description of the announcement

Hello @robotology/everyone  👋

The [iCub project](https://icub.iit.it/) is happy to announce our **Software Distro 2024.11.1** 📦

This is a major release that comprises **binary packages** as well. Therefore, follow the [instructions][1] available in our documentation to upgrade your system and/or download the new dependencies.

You can find the tags composing this distro in our [Software Versioning Table][2].

## Detailed context

This distro is shipped with several updates and fixes, including the ones listed hereafter:

#### 🔘 `icub-firmware-models`: MBD for AMC-FOC
We extended our functional architecture responsible for generating the motor control's embedded code of the AMC-* family boards so that we now cover the case of the 2 PMSM motors controlled by the AMC-FOC board. Kudos to @mfussi66!
For details, see:
- https://github.com/robotology/icub-firmware-models/pull/94

#### 🔘 ROS 2 support via yarp-devices-ros2 released and available in superbuild

The [`yarp-devices-ros2` repository](https://github.com/robotology/yarp-devices-ros2) that contains [YARP Network Wrapper Server (NWS) and Network Wrapper Client (NWC)](https://www.yarp.it/latest//group__nws__and__nwc__architecture.html) devices that read and write data on ROS 2 topic. The repo has seen its first release, and it is not included as part of the robotology distro packages.

To install `yarp-devices-ros2` when you build the software from source, you can use the new robotology-superbuild option `ROBOTOLOGY_USES_ROS2` that enable the compilation of the `yarp-devices-ros2` repo. At the moment, no conda binary package is available for the `yarp-devices-ros2` package, but it is planned to add one in the next months.

Refer to the [`yarp-devices-ros2` documentation](https://github.com/robotology/yarp-devices-ros2) documentation for details on how to use the devices contained in it.

#### 🔘 New C++ library `qpsolvers-eigen` that abstracts away QP solvers

For a long time, the robotology distro included the [`osqp-eigen` C++ library](https://github.com/robotology/osqp-eigen) to solve [Quadratic Programming optimization problems](https://en.wikipedia.org/wiki/Quadratic_programming) in C++. This release of the robotology distro includes the  [`qpsolvers-eigen` C++ library](https://github.com/ami-iit/qpsolvers-eigen) library that provides an API similar to `osqp-eigen`, but permitting to easily switch between two different solvers [`osqp`](https://osqp.org/) or [`proxqp`](https://github.com/Simple-Robotics/proxsuite).

If you build from source you can install `qpsolvers-eigen`  by enabling the `ROBOTOLOGY_ENABLE_DYNAMICS` CMake option of the robotology-superbuild, or if you use conda binaries you can just install the `qpsolvers-eigen` package from the `conda-forge` channel.

For examples of usage please check the [`qpsolvers-eigen` documentation](https://github.com/ami-iit/qpsolvers-eigen?#%EF%B8%8F-how-to-use-the-library) or if you want to migrate from `osqp-eigen` check the [migration guide from `osqp-eigen` to `qpsolvers-eigen`](https://github.com/ami-iit/qpsolvers-eigen/blob/v0.1.0/docs/migrate_from_osqp_eigen.md).

#### 🔘 `creo2urdf`: added vcpkg manifest
In:
- https://github.com/icub-tech-iit/creo2urdf/pull/116

we facilitated the installation from sources of `creo2urdf` adding the vcpkg manifest.
This json file allow to install the correct vcpkg dependencies while configuring with `cmake`.

#### 🔘 Improved ergocub models meshes

In:
- https://github.com/icub-tech-iit/ergocub-software/pull/266

we added a set of new meshes made by @Supermonkey-design for the ergoCub urdf models.
Here is how they look like:
| **SN000** |   **SN001/2** |
|----------|:---------:|
| ![image](https://github.com/user-attachments/assets/ef9751a0-6739-4053-9ebc-ffc0bcb0dbca) |  ![image](https://github.com/user-attachments/assets/3f91fce8-26fd-4f31-b064-b32560665d9e) |

#### 🔘 `icub-main`: raw values interface
In :
- https://github.com/robotology/icub-main/pull/983 

we introduce a new interface though for facilitating low-level debugging operation. It basically allows users to stream to yarp ports whatever raw values they desire, with the prerogative of updating the firmware accordingly.
Moreover, it is connected to the `TelemetryDeviceDumper` plugin, so that, those raw values can be saved to a `MATLAB` `.m` file.

#### 🔘 ROS 1 compatibility deprecated

As the last supported ROS 1 distribution (ROS Noetic Ninjemys) is reaching its End of Life (EOL) on [May 2025](http://wiki.ros.org/Distributions), this release is deprecating the existing form of ROS 1 support in the distribution, including:
* All the YARP [Network Wrapper Server (NWS) and Network Wrapper Client (NWC)](https://www.yarp.it/latest//group__nws__and__nwc__architecture.html) devices that use ROS 1, contained and listed in https://github.com/robotology/yarp-devices-ros,
* All the YARP carriers used to support ROS 1, including `tcpros` and `xmlrpc` carriers, contained in https://github.com/robotology/yarp-ros,
* All the YARP idl generators related to the support for generating YARP serializable classes out of ROS 1's `.msg` files,
* All the `yarpserver` capabilities for interacting with `roscore`, including the `yarpserver --ros` option.

All these functionalities is deprecated by the distro release 2024.11.0, and will be removed in the distro release 2025.02.0 .

The current migration path to ROS 2 is suggested:
* For ROS-compatible NWS and NWC devices, the ROS 2-power devices provided in [`yarp-devices-ros2`](https://github.com/robotology/yarp-devices-ros2) should be used instead of the deprecated ROS 1-enabled devices that used to be contained in `yarp` and are now contained in [`yarp-devices-ros`](https://github.com/robotology/yarp-devices-ros). Compilation of the `yarp-devices-ros2` repository can be enabled in the `robotology-superbuild` by enabling the `ROBOTOLOGY_USES_ROS2` CMake option.
* For reading or writing data for ROS 2 topics, it is suggested to use the official ROS 2 C++ api to read and write topics, see https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Topics/Understanding-ROS2-Topics.html .
* For YARP serialible types, it is recommended to use `thrift` IDL definitions instead of `rosmsg` definitions, see https://yarp.it/latest/idls.html .

For any question or doubt related to ROS 1 deprecation, please post your question at https://github.com/orgs/robotology/discussions/categories/q-a .

#### 🔘 iCub3 urdf models removed

In:
- https://github.com/robotology/icub-models-generator/pull/281
- https://github.com/robotology/icub-models/pull/243

we removed the iCub3 urdf models since this robot became `iRonCubMk3` and `iRonCubMk3`'s urdf is maintained in another repository.

[1]: https://icub-tech-iit.github.io/documentation/sw_installation/
[2]: https://icub-tech-iit.github.io/documentation/sw_versioning_table/#table

**Comment by Nicogene** [comment]
Hi @robotology/everyone,

we just released the [**Distro `2024.11.1`**](https://icub-tech-iit.github.io/documentation/sw_versioning_table/2024.11.1/) since I forgot to tag some repos under my maintenance, sorry for the inconvenience. I hope that this has not created any trouble in your workflow!

**Comment by traversaro** [comment]
Thanks to @antonellopaolino and @davidegorbani we noticed that there was an undocumented breaking change in BlockFactory 1.0.0 that was updated in 2024.11.1 . If you are seeing BlockFactory-related compilation errors on your downstream code, please check the migration guide in https://github.com/robotology/blockfactory/releases/tag/v1.0.0 .

**Comment by Nicogene** [comment]
Hi @robotology/everyone,

we just released the [Distro `2024.11.2`](https://icub-tech-iit.github.io/documentation/sw_versioning_table/2024.11.2), that incorporate important fixes such as:
- https://github.com/robotology/ycm-cmake-modules/pull/468
-  https://github.com/robotology/icub-firmware/pull/551
